elide-core/src/main/java/com/yahoo/elide/parsers/expression/PermissionToFilterExpressionVisitor.java

```diff
@@ -60,9 +73,26 @@ public FilterExpression visitOR(ExpressionParser.ORContext ctx) {
             return NO_EVALUATION_EXPRESSION;
         }
 
-        //This special case require future reconsideration.
-        if (left == USER_CHECK_EXPRESSION || right == USER_CHECK_EXPRESSION) {
-            return USER_CHECK_EXPRESSION;
+        if (left instanceof UserCheckExpression || right instanceof UserCheckExpression) {
```

Comment:
This would become:
```
if (left == FALSE_USER_CHECK_EXPRESSION) {
  return right
} else if (right == FALSE_USER_CHECK_EXPRESSION) {
  return left;
}
```